sessionctx/variable/session.go

```diff
@@ -1478,7 +1488,7 @@ func (connInfo *ConnectionInfo) IsSecureTransport() bool {
 }
 
 // NewSessionVars creates a session vars object.
-func NewSessionVars() *SessionVars {
+func NewSessionVars(hctx HookContext) *SessionVars {
```

Comment:
Does it necessary to change the definition of NewSessionVars, which introduced so many changes.

Reply:
Sadly, these changes have been merged into master in https://github.com/pingcap/tidb/pull/38227. @morgo Provided a better way to work around this :+1: . I'll revert these changes after https://github.com/pingcap/tidb/pull/38182 merged :smiley: